app/api/v1/admin/patients.rb

```diff
@@ -0,0 +1,109 @@
+module V1
+  module Admin
+    class Patients < API
+      patients_crud_url = '/api/v1/admin/patients'
+      resources :patients do
+        desc 'Return all patients'
+        get do
+          patients = Patient.all
+          present patients, with: Entities::Patient
+        end
+
+        desc 'Return a specific patient'
+        route_param :id do
+          get do
+            patient = Patient.find(params[:id])
+            present patient, with: Entities::Patient
+          rescue ActiveRecord::RecordNotFound
+            error!({ error_code: 404, error_message: 'Patient not found' })
+          end
+        end
+
+        desc 'Create a new patient'
+        params do
+          requires :address, :date_of_birth, :email, :full_name, :notes, :telephone_number,
+                   :passport_id, :allergies_additional, :allergies
+        end
+        post do
+          patient = Patient.create(
+            address: params[:address],
+            date_of_birth: params[:date_of_birth],
+            email: params[:email],
+            full_name: params[:full_name],
+            notes: params[:notes],
+            telephone_number: params[:telephone_number],
+            passport_id: params[:passport_id],
+            allergies_additional: params[:allergies_additional]
+          )
+          params[:allergies].split(/,/).each do |allergy_id|
+            patient.allergies << Allergy.find(allergy_id)
+          rescue ActiveRecord::RecordNotFound
+            error!({ error_code: 400, error_message: 'Invalid data of allergies' })
+          end
+          if patient.valid?
+            patient.save
```

Comment:
it's a huge method, try to create methods with less than 10 lines - it's more in Ruby style

for me, it looks like a separate class, try to avoid fat controllers/models

create a service class for those actions or any other pattern